src/command_modules/azure-cli-subscription/azure/cli/command_modules/subscription/_params.py

```diff
@@ -0,0 +1,17 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.command_modules.subscription._completers import get_offer_type_completion_list
+
+
+# pylint: disable=line-too-long
+def load_arguments(self, _):
+    with self.argument_context('subscriptiondefinition show') as c:
+        c.argument('subscription_definition_name', options_list=['--name', '-n'], required=False, help='Name of the subscription definition to show.')
+
+    with self.argument_context('subscriptiondefinition create') as c:
+        c.argument('name', options_list=['--name', '-n'], required=True, help='Name of the subscription definition.')
+        c.argument('offer_type', options_list=['--offer_type', '-ot'], required=True, help='The subscription\'s offer type.', completer=get_offer_type_completion_list)
```

Comment:
You cannot have a two-character short option.  If anything there shouldn't be a short option, but if you really want one, `-t` would be the most appropriate.  

You cannot use underscore in you long options. `--offer_type` should be `--offer-type`.

Just use `arg_type=get_enum_type(...)` here instead of a completer.

Finally, you don't need/should not provide required=True/False. The CLI will infer that from your command signature based on whether you provided a default value.

Reply:
Fixed.